python/src/aiconfig/editor/server/server.py

```diff
@@ -49,137 +88,120 @@ def home():
     state = get_server_state(app)
-    state.aiconfig = AIConfigRuntime.create()  # type: ignore
-    return HttpResponseWithAIConfig(message="Created new AIConfig", aiconfig=state.aiconfig).to_flask_format()
+    aiconfig = safe_run_aiconfig_static_method(MethodName("create"), OpArgs({}), AIConfigRuntime)
+    match aiconfig:
+        case Ok(aiconfig_ok):
+            state.aiconfig = aiconfig_ok
+            return HttpResponseWithAIConfig(message="Created new AIConfig", aiconfig=aiconfig_ok).to_flask_format()
+        case Err(e):
+            return HttpResponseWithAIConfig(message=f"Failed to create AIConfig: {e}", code=400, aiconfig=None).to_flask_format()
 
 
 @app.route("/api/run", methods=["POST"])
 async def run() -> FlaskResponse:
     state = get_server_state(app)
+    aiconfig = state.aiconfig
     request_json = request.get_json()
+
+    def _op(aiconfig: AIConfigRuntime, op_args: OpArgs) -> Result[None, str]:
+        method_name = MethodName("run")
+        return safe_run_aiconfig_method(aiconfig, method_name, op_args)
+
     prompt_name = request_json.get("prompt_name", None)
     stream = request_json.get("stream", True)
     LOGGER.info(f"Running prompt: {prompt_name}, {stream=}")
     inference_options = InferenceOptions(stream=stream)
-    try:
-        result = await state.aiconfig.run(prompt_name, options=inference_options)  # type: ignore
-        LOGGER.debug(f"Result: {result=}")
-        return HttpResponseWithAIConfig(
-            message="Done",
-            aiconfig=state.aiconfig,
-        ).to_flask_format()
-    except Exception as e:
-        err: Err[str] = core_utils.ErrWithTraceback(e)
-        LOGGER.error(f"Failed to run: {err}")
-        return HttpResponseWithAIConfig(
-            message=f"Failed to run: {err}",
-            code=400,
-            aiconfig=None,
-        ).to_flask_format()
+
+    op_args = Ok(OpArgs({"prompt_name": prompt_name, "options": inference_options}))
+
+    return run_aiconfig_operation_with_op_args(aiconfig, "run", _op, op_args)
 
 
 @app.route("/api/add_prompt", methods=["POST"])
 def add_prompt() -> FlaskResponse:
-    state = get_server_state(app)
-    request_json = request.get_json()
-    try:
-        LOGGER.info(f"Adding prompt: {request_json}")
-        state.aiconfig.add_prompt(**request_json)  # type: ignore
-        return HttpResponseWithAIConfig(
-            message="Done",
-            aiconfig=state.aiconfig,
-        ).to_flask_format()
-    except Exception as e:
-        err: Err[str] = core_utils.ErrWithTraceback(e)
-        LOGGER.error(f"Failed to add prompt: {err}")
-        return HttpResponseWithAIConfig(
-            message=f"Failed to add prompt: {err}",
-            code=400,
-            aiconfig=None,
-        ).to_flask_format()
+    method_name = MethodName("add_prompt")
+    signature: dict[str, Type[Any]] = {"prompt_name": str, "prompt_data": Prompt}
 
+    return _run_method_endpoint(method_name, signature)
```

Comment:
nit, it's really confusing when to use safe_run_aiconfig_method , run_aiconfig_operation_with_op_args, run_aiconfig_operation_with_request_json, or _run_method_endpoint

They all sound pretty similar and it's hard to make sense of the difference in the code. Can we reduce this to one helper (or two at most if needed, but with clear documentation on when to use one or the other) that can be used everywhere here for calling aiconfig methods?

Reply:
Point taken. It's not immediately obvious to me how to simplify this without a lot of duplication and refactoring. Let's fix after. I'll document for now.

Reply:
Ok, I got it down to 2 with a little extra duplication and added docstrings to those 2.